contracts/src/libraries/XTypes.sol

```diff
@@ -7,60 +7,89 @@ pragma solidity ^0.8.12;
  *      types mirror those defined in omni/lib/xchain/types.go.
  */
 library XTypes {
-    struct Msg {
-        /// @dev Chain ID of the source chain
+    /**
+     * @notice The full context of an xmsg. This matches the XMsg type used throughout Omni's
+     *         cross-chain messaging protocol. MsgFull is used to contrsuct and verify XSubmission
+     *         merkle trees / proofs.
+     * @custom:field sourceChainId  Chain ID of the source chain
+     * @custom:field destChainId    Chain ID of the destination chain
+     * @custom:field streamOffset   Monotonically incremented offset of Msg in source -> dest Stream
+     * @custom:field sender         msg.sender of xcall on source chain
+     * @custom:field to             Target address to call on destination chain
+     * @custom:field data           Data to provide to call on destination chain
+     * @custom:field gasLimit       Gas limit to use for call execution on destination chain
+     */
+    struct MsgFull {
```

Comment:
Since this is the "true" `XMsg`, and the one below is a trimmed version, seems we should call this `XMsg` and the trimmed one something like `MsgShort`

But we can still keep the `xmsg.sender` in `Xapp` by setting xmsg to a `MsgShort` type

Reply:
Yeah I had XMsgTrim at first.

But wanted the user facing type to be XTypes.Msg 

There might be better ways to get that result tho.